File: app/src/main/java/com/zurichat/app/ui/organizations/localdatabase/TypeConverters/StringListTypeConverter.java
package com.zurichat.app.ui.organizations.localdatabase.TypeConverters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.zurichat.app.ui.fragments.model.Data;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.List;

public class StringListTypeConverter {
    Gson gson = new Gson();

    @TypeConverter
    public List<String> storedStringToStringList(String value) {
        if (value == null) {
            return Collections.emptyList();
        }
        Type listType = new TypeToken<List<Data>>() {}.getType();
        return gson.fromJson(value, listType);
    }

    @TypeConverter
    public String stringListToStoredString(List<String> emojiList) {
        return gson.toJson(emojiList);
    }
}

File: app/src/main/java/com/zurichat/app/ui/organizations/localdatabase/TypeConverters/StringListTypeConverter.kt
package com.zurichat.app.ui.organizations.localdatabase.TypeConverters

import androidx.room.TypeConverter
import com.google.gson.Gson

class StringListTypeConverter {
    @TypeConverter
    fun imagesToJson(images:List<String>): String = Gson().toJson(images)

    @TypeConverter
    fun jsonToImages(value: String) = Gson().fromJson(value, Array<String>::class.java).toList()
}